$NetBSD: patch-lib-src_Makefile.in.in,v 1.1 2017/11/15 14:52:02 hauke Exp $

Add missing DESTDIR prefixes

Install internal build scripts as scripts

Prefix {c,e}tags to avoid conflicts

--- lib-src/Makefile.in.in.orig	2025-01-06 19:10:12.962500316 +0000
+++ lib-src/Makefile.in.in
@@ -242,9 +242,9 @@ do-blessmail: $(blessmail)
 ${archlibdir}: all
 	@echo; echo "Installing utilities run internally by XEmacs."
 	./make-path $(DESTDIR)${archlibdir}
-	if test "`(cd ${archlibdir} && $(pwd))`" != "`$(pwd)`"; then \
+	if test "`(cd $(DESTDIR)${archlibdir} && $(pwd))`" != "`$(pwd)`"; then \
 	  for f in ${PRIVATE_INSTALLABLE_EXES}; do \
-	    (cd .. && $(INSTALL_PROGRAM) lib-src/$$f $(DESTDIR)${archlibdir}/$$f) ; \
+	    (cd .. && $(INSTALL_SCRIPT) lib-src/$$f $(DESTDIR)${archlibdir}/$$f) ; \
 	  done ; \
 	fi
 	if test "`(cd ${archlibdir} && $(pwd))`" \
@@ -257,10 +257,20 @@ ${archlibdir}: all
 install: ${archlibdir}
 	@echo; echo "Installing utilities for users to run."
 	for file in ${PUBLIC_INSTALLABLE_EXES} ; do \
-	  (cd .. && $(INSTALL_PROGRAM) lib-src/$${file} $(DESTDIR)${bindir}/$${file}) ; \
+          if [ $${file} = etags ] || [ $${file} = ctags ] || [ $${file} = b2m ] ; \
+            then \
+              (cd .. && $(INSTALL_PROGRAM) lib-src/$${file} $(DESTDIR)${bindir}/xemacs-$${file}) ; \
+            else \
+              (cd .. && $(INSTALL_PROGRAM) lib-src/$${file} $(DESTDIR)${bindir}/$${file}) ; \
+          fi \
 	done
 	for file in ${PUBLIC_INSTALLABLE_SCRIPTS} ; do \
-	  (cd .. && $(INSTALL_SCRIPT) ${srcdir}/$${file} $(DESTDIR)${bindir}/$${file}) ; \
+          if [ $${file} = rcs-checkin ] || [ $${file} = send-pr ] ; \
+            then \
+              (cd .. && $(INSTALL_SCRIPT) ${srcdir}/$${file} $(DESTDIR)${bindir}/xemacs-$${file}) ; \
+            else \
+              (cd .. && $(INSTALL_SCRIPT) ${srcdir}/$${file} $(DESTDIR)${bindir}/$${file}) ; \
+          fi \
 	done
 
 uninstall:
